editor/sidebar/post-visibility/index.js

```diff
@@ -71,6 +74,45 @@ class PostVisibility extends Component {
 		this.setState( { opened: false } );
 	}
 
+	handleBlur( event ) {
+		if ( this.state.opened && event.relatedTarget !== null ) {
+			const wrapper = findDOMNode( this );
+			const toggle = wrapper.querySelector( '.editor-post-visibility__toggle' );
+			const dialog = wrapper.querySelector( '.editor-post-visibility__dialog' );
+
+			if ( ! dialog.contains( event.relatedTarget ) ) {
+				this.setState( { opened: false } );
+				toggle.focus();
+			}
+		}
+	}
+
+	handleKeyDown( event ) {
+		if ( this.state.opened && event.keyCode === ESCAPE ) {
+			const wrapper = findDOMNode( this );
+			const toggle = wrapper.querySelector( '.editor-post-visibility__toggle' );
+
+			if ( event.target === toggle ) {
+				return;
+			}
+
+			event.preventDefault();
+			event.stopPropagation();
+			this.setState( { opened: false } );
+			toggle.focus();
+		}
+	}
+
+	componentDidMount() {
+		const node = findDOMNode( this );
+		node.addEventListener( 'keydown', this.handleKeyDown, false );
```

Comment:
Was there a reason we couldn't apply an `onKeyDown` prop to the rendered `<div />` ?